polaris-core/src/main/java/org/apache/polaris/core/config/RelationalJdbcConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.config;
+
+import io.smallrye.config.ConfigMapping;
+import java.util.Map;
+import java.util.Optional;
+
+@ConfigMapping(prefix = "polaris.relation.jdbc.datasource")
+public interface RelationalJdbcConfiguration {
+  /** realmId to configured Datasource name mapping. */
+  Map<String, String> realm();
```

Comment:
```suggestion
  Map<String, String> realms();
```

Also, you probably need to annotate this with `@WithParentName` and maybe `@WithUnnamedKey`, depending on how you want to handle the "default" Jdbc configuration.